### PR TITLE
initialize library code

### PR DESCRIPTION
**Pull request by maZahaca**

### CODE REVIEW

File: lib/parser.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/*!
+ * parser
+ * Copyright(c) 2015 Andrew Reddikh <andrew@reddikh.com>
+ * MIT Licensed
+ */
+
+var used = []
+  , exports = module.exports = {};
```
Comment (jifeon) [inline lib/parser.js]: you are using leading comma only here

File: lib/parser/PaginateableParser.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
+try {
+    var RedParser = require('./RedParser');
+} catch (e) {
```
Comment (jifeon) [inline lib/parser/PaginateableParser.js]: it's really bad practice to ignore errors this way

File: lib/parser.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+/*!
+ * Parser version
+ */
+
+exports.version = '0.1.0';
+
+var RedParser = require('./parser/RedParser.js');
```
Comment (jifeon) [inline lib/parser.js]: redundant variable

File: lib/parser/PaginateableParser.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+}
+
+/**
+ * @class
+ * @extends RedParser
+ */
+var PaginateableParser = function () {
+    //RedParser.call(this);
```
Comment (jifeon) [inline lib/parser/PaginateableParser.js]: remove all comments

File: lib/parser/PaginateableParser.js [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+
+if (typeof debug !== 'function') {
+    var debug = function () {
+    };
+}
+
+PaginateableParser.prototype = Object.create(RedParser.prototype);
+PaginateableParser.prototype.constructor = PaginateableParser;
+
+PaginateableParser.prototype._TYPES.PAGINATION = {
```
Comment (jifeon) [inline lib/parser/PaginateableParser.js]: using `Object.assign` would look better:

``` js
Object.assign(PaginateableParser.prototype, {
  _TYPES: {
  },

  paginate: function() {},

  ...
});
```

File: index.js [hunk 6]
```diff
@@ -0,0 +1 @@
+module.exports = require('./lib/parser');
```
Comment (jifeon) [inline index.js]: you can just specify it in package.json as `main`

**Comment by jifeon** [comment]
@maZahaca do not leave dead branches

**Comment by maZahaca** [comment]
:boar: okay

**Pull request merged**